python/tchannel/req_handler.py

```diff
@@ -0,0 +1,69 @@
+import collections
+
+EndPoint = collections.namedtuple('EndPoint', ['handler', 'opts'])
+
+
+class RequestHandler(object):
+
+    def __init__(self):
+        pass
+
+    def handle_request(self, context, conn):
+        raise NotImplementedError()
+
+
+class TChannelRequestHandler(RequestHandler):
+    def __init__(self):
+        self.endpoints = {}
+
+    def handle_request(self, context, conn):
+        """Handle incoming request
+
+        :param context: incoming message context
+        :param conn: incoming connection
+        """
+        request = TChannelRequest(context, conn)
+        endpoint = self._find_endpoint(request.method)
+        if endpoint is not None:
+            response = TChannelResponse(conn)
+            endpoint.handler(request, response, endpoint.opts)
+            response.finish()
+
+    def route(self, rule, **opts):
+        def decorator(handler):
+            self.register_handler(rule, handler, **opts)
+            return handler
+
+        return decorator
+
+    def register_handler(self, rule, handler, **opts):
+        self.endpoints[rule] = EndPoint(handler=handler, opts=opts)
+
+    def _find_endpoint(self, rule):
+        return self.endpoints.get(rule, None)
+
+
+class TChannelRequest(object):
+    """ TChannel Request Wrapper """
```

Comment:
Nit: Generally, don't leave an extra space between the quotes and the description.